search-spec/openapi.yml

```diff
@@ -0,0 +1,229 @@
+openapi: '3.0.0'
+info:
+  version: 0.0.1
+  title: UP42 Tasking
+  license:
+    name: MIT
+servers:
+  - url: /v1
+paths:
+  /search:
+    post:
+      summary: Get possible tasking options for spatial temporal query
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SearchQuery"
+      responses:
+        200:
+          description: A paged array of tasking options
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SlotCollections"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+components:
+  schemas:
+    SlotCollections:
+      type: array
+      items:
+        type: object
+        properties:
+          slotCollection:
+            type: object
+            description: "A slotcollection is an offerings of slots which share the same tasking properties"
+            properties:
+              timestamp:
+                type: string
+                format: date-time
+                example: '2017-07-21T17:32:28Z'
+              taskingProperties:
+                $ref: '#/components/schemas/TaskingProperties'
+              productConfiguration:
+                $ref: '#/components/schemas/ProductConfiguration'
+              priority:
+                $ref: '#/components/schemas/Priority'
+              price:
+                $ref: '#/components/schemas/Price'
+              lifecycle:
+                type: array
+                example: [ "FEASIBILITY", "PRICE", "DELIVERY" ]
+                items:
+                  type: string
```

Comment:
Lifecycle stages are going to be critical in different places in this API, aren't they?

I'd probably define a separate schema component to reuse for them. And the type of that component would be ~JOHN CENA~ ... ghm... `enum` I mean.

See https://swagger.io/docs/specification/data-models/enums/ .